lib/actions-oclif/os/configure.ts

```diff
@@ -0,0 +1,430 @@
+			OsConfigureCmd,
+		);
+		// Prefer options.application over options.app
+		options.application = options.application || options.app;
+		options.app = undefined;
+
+		await validateOptions(options);
+
+		const devInit = await import('balena-device-init');
+		const balena = (await import('balena-sdk')).fromSharedOptions();
+		const fs = await import('mz/fs');
+		const { generateDeviceConfig, generateApplicationConfig } = await import(
+			'../../utils/config'
+		);
+		const helpers = await import('../../utils/helpers');
+		let app: BalenaSdk.Application | undefined;
+		let device: BalenaSdk.Device | undefined;
+		let deviceTypeSlug: string;
+
+		if (options.device) {
+			device = await balena.models['device'].get(options.device);
+			deviceTypeSlug = device.device_type;
+		} else {
+			app = await balena.models['application'].get(options.application!);
+			await checkDeviceTypeCompatibility(balena, options, app);
+			deviceTypeSlug = options['device-type'] || app.device_type;
+		}
+
+		const deviceTypeManifest = await helpers.getManifest(
+			params.image,
+			deviceTypeSlug,
+		);
+
+		let configJson: import('../../utils/config').ImgConfig | undefined;
+		if (options.config) {
+			const rawConfig = await fs.readFile(options.config, 'utf8');
+			configJson = JSON.parse(rawConfig);
+		}
+
+		const answers: Answers = await askQuestionsForDeviceType(
+			deviceTypeManifest,
+			options,
+			configJson,
+		);
+		if (options.application) {
+			answers.deviceType = deviceTypeSlug;
+		}
+		answers.version =
+			options.version ||
+			(await getOsVersionFromImage(params.image, deviceTypeManifest, devInit));
+
+		if (_.isEmpty(configJson)) {
+			if (device) {
+				configJson = await generateDeviceConfig(
+					device as DeferredDevice,
+					options['device-api-key'],
+					answers,
+				);
+			} else {
+				configJson = await generateApplicationConfig(app!, answers);
+			}
+		}
+
+		console.info('Configuring operating system image');
+
+		await helpers.osProgressHandler(
+			await devInit.configure(
+				params.image,
+				deviceTypeManifest,
+				configJson || {},
+				answers,
+			),
+		);
+	}
+}
+
+async function validateOptions(options: FlagsDef) {
+	if (process.platform === 'win32') {
+		throw new ExpectedError(stripIndent`
+			Unsupported platform error: the 'balena os configure' command currently requires
+			the Windows Subsystem for Linux in order to run on Windows. It was tested with
+			the Ubuntu 18.04 distribution from the Microsoft Store. With WSL, a balena CLI
+			release for Linux (rather than Windows) should be installed: for example, the
+			standalone zip package for Linux. (It is possible to have both a Windows CLI
+			release and a Linux CLI release installed simulatenously). For more information
+			on WSL and the balena CLI installation options, please check:
+			- https://docs.microsoft.com/en-us/windows/wsl/about
+			- https://github.com/balena-io/balena-cli/blob/master/INSTALL.md
+		`);
```

Comment:
I've created a separate CLI issue for this:
#1509 ["balena os configure" hangs on Windows (not WSL)](https://github.com/balena-io/balena-cli/issues/1509)